src/DatabaseProvider/index.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import React from 'react'
+import Database from '../Database'
+
+const { Provider, Consumer } = React.createContext({})
+
+export type Props = {
+  database: Database,
+  children: React$Element<any>,
+}
+
+/**
+ * Database provider to create the database context
+ * to allow child components to consume the database without prop drilling
+ */
+function DatabaseProvider({ children, database }: Props): React$Element<any> {
```

Comment:
Flow isn't happy about that:

```bash
Cannot return <Provider /> because:
 • property value is missing in Props [1] but exists in object type [2] in the first argument of type argument
   ElementType [3].
 • inexact React.Element [4] is incompatible with exact $Iterable [5] in the return value of type argument
   ElementType [3].
 • inexact React.Element [4] is incompatible with exact React.Portal [5] in the return value of type argument
   ElementType [3].
 • inexact React.Element [4] is incompatible with exact boolean [5] in the return value of type argument
   ElementType [3].
 • inexact React.Element [4] is incompatible with exact null [5] in the return value of type argument
   ElementType [3].
 • inexact React.Element [4] is incompatible with exact number [5] in the return value of type argument
   ElementType [3].
 • inexact React.Element [4] is incompatible with exact string [5] in the return value of type argument
   ElementType [3].
 • in property children of the first argument of type argument ElementType [3]:
    • Either null or undefined [6] is incompatible with null [7].
    • Or property @@iterator is missing in null or undefined [6] but exists in $Iterable [8].
 • in property children of the first argument of type argument ElementType [3]:
    • Either undefined [6] is incompatible with null [7].
    • Or property @@iterator is missing in undefined [6] but exists in $Iterable [8].
 • a callable signature is missing in statics of React.Component [9] but exists in function [10] in type argument
   ElementType [3].

            src/DatabaseProvider/index.js
 [1][4][10]  16│ function DatabaseProvider({ children, database }: Props): React$Element<typeof DatabaseProvider> {
             17│   if (!database) {
             18│     throw new Error('You must supply a database prop to the DatabaseProvider')
             19│   }
             20│   return <Provider value={database}>{children}</Provider>
             21│ }
             22│
             23│ export { Consumer as DatabaseConsumer }

            /private/tmp/flow/flowlib_2cdfc57b/react.js
        [7]  14│   | null
               :
        [8]  20│   | Iterable<?React$Node>;
               :
        [5] 138│   (props: Props, context: any): React$Node,
               :
        [3] 173│ declare type React$Element<+ElementType: React$ElementType> = {|
               :
  [9][2][6] 199│   Provider: React$ComponentType<{ value: T, children?: ?React$Node }>,
```

I've also tried
```javascript
// doesn't like
React$Element<DatabaseProvider>

// it doesn't mind
React$Element<typeof Provider>
```
😅